### PR TITLE
FS-3076 copilot aws deploy through workflow

### PR DESCRIPTION
**Pull request by robk-dluhc**
https://dluhcdigital.atlassian.net/browse/FS-3074

### Change description
Adding account store connected database
Adding copilot deploy functionality (default inactive)
Adding all copilot manifest configuration files

- [n/a] Unit tests and other appropriate tests added or updated
- [X] README and other documentation has been updated / added (if needed)
- [X] Commit messages are meaningful and follow good commit message guidelines (e.g. "FS-XXXX: Add margin to nav items preventing overlapping of logo")

### How to test
Run Deploy Action Workflow with the copilot input set to "True"

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -34,3 +47,45 @@ jobs:
...
+
+    - name: Get current date
+      id: currentdatetime
+      run: echo "datetime=$(date +'%Y%m%d%H%M%S')" >> $GITHUB_OUTPUT
+
+    - name: configure aws credentials
+      uses: aws-actions/configure-aws-credentials@v2
+      with:
+        role-to-assume: arn:aws:iam::${{ secrets.AWS_ACCOUNT }}:role/GithubCopilotDeploy
+        role-session-name: ACCOUNT_STORE_COPILOT_${{ steps.currentdatetime.outputs.datetime }}
```
Comment (aaronwilliamsv1) [inline .github/workflows/deploy.yml:71]: This isn't the account store :D

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -34,3 +47,45 @@ jobs:
...
+    - name: Inject Git SHA into manifest
+      run: |
+        yq -i '.variables.GITHUB_SHA = "${{ github.sha }}"'  copilot/fsd-application-store/manifest.yml
+
+    - name: Inject replacement image into manifest
+      run: |
+        yq -i '.image.location = "ghcr.io/communitiesuk/funding-service-design-application-store:${{ github.ref_name }}"'  copilot/fsd-application-store/manifest.yml
+
+    - name: Run database migrations
+      run: scripts/migration-task-script.py ${{ inputs.environment || 'test' }}
```
Comment (aaronwilliamsv1) [inline .github/workflows/deploy.yml:87]: Not sure what script this is referencing

Comment (aaronwilliamsv1) [reply .github/workflows/deploy.yml:87]: Ignore me, spotted this below

File: scripts/migration-task-script.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env python3
+import re
+import subprocess
+import sys
+
+command_to_run = "flask db upgrade"
```
Comment (aaronwilliamsv1) [inline scripts/migration-task-script.py:6]: This matches up with what runs under https://github.com/communitiesuk/funding-service-design-application-store/blob/main/scripts/run_migrations_paas.py#L16

**Review by aaronwilliamsv1 — CHANGES_REQUESTED** [review]
Application Store database add on missing

File: copilot/fsd-application-store/addons/fsd-application-store-cluster.yml [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+    Value:
+      !Sub
+      - "postgres://${USERNAME}:${PASSWORD}@${HOSTNAME}:${PORT}/${DBNAME}"
+      - USERNAME: !Join [ "",  [ '{{resolve:secretsmanager:', !Ref fsdapplicationstoreclusterAuroraSecret, ":SecretString:username}}" ]]
+        PASSWORD: !Join [ "",  [ '{{resolve:secretsmanager:', !Ref fsdapplicationstoreclusterAuroraSecret, ":SecretString:password}}" ]]
+        HOSTNAME: !Join [ "",  [ '{{resolve:secretsmanager:', !Ref fsdapplicationstoreclusterAuroraSecret, ":SecretString:host}}" ]]
+        PORT: !Join [ "",  [ '{{resolve:secretsmanager:', !Ref fsdapplicationstoreclusterAuroraSecret, ":SecretString:port}}" ]]
+        DBNAME: !Join [ "",  [ '{{resolve:secretsmanager:', !Ref fsdapplicationstoreclusterAuroraSecret, ":SecretString:dbname}}" ]]
+
+  fsdapplicationstoreclusterSecret: # injected as FSDACCOUNTSTORECLUSTER_SECRET environment variable by Copilot.
```
Comment (aaronwilliamsv1) [inline copilot/fsd-application-store/addons/fsd-application-store-cluster.yml:135]: Account

**Comment by robk-dluhc** [comment]
Moved branch

**Comment by aaronwilliamsv1** [comment]
Migrated to FS-3074

**Pull request closed without merging**